src/v/kafka/data/replicated_partition.cc

```diff
@@ -354,8 +360,10 @@ raft::replicate_stages replicated_partition::replicate(
           ss::make_ready_future<result<raft::replicate_result>>(
             make_error_code(kafka::error_code::invalid_topic_exception))};
     }
-
-    auto res = _partition->replicate_in_stages(batch_id, std::move(rdr), opts);
+    chunked_vector<model::record_batch> batches;
+    batches.push_back(std::move(batch));
```

Comment:
FYI only: for these situations `std::ranges::single_view` can work with `chunked_vector::from` if you find it neater.